spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala

```diff
@@ -297,5 +306,104 @@ object SparkTableUtil {
       )
     }
   }
+
+  private def buildManifest(conf: SerializableConfiguration,
+      sparkDataFiles: Seq[SparkDataFile],
+      partitionSpec: PartitionSpec,
+      basePath: String): Iterator[Manifest] = {
+    if (sparkDataFiles.isEmpty) {
+      Seq.empty.iterator
+    }
+
+    val io = new HadoopFileIO(conf.get())
+    val ctx = TaskContext.get()
+    val location = new Path(basePath,
+      s"stage-${ctx.stageId()}-task-${ctx.taskAttemptId()}-manifest")
+    val outputFile = io.newOutputFile(FileFormat.AVRO.addExtension(location.toString))
+    val writer = ManifestWriter.write(partitionSpec, outputFile)
+    try {
+      sparkDataFiles.foreach { file =>
+        writer.add(file.toDataFile(partitionSpec))
```

Comment:
What will be happened if an exception is raised here?

Reply:
The expected behaviour is spark retry the task. If one of the tasks fail eventually, the job is failed and manifests are discarded from the driver as well.  As a result, the partial result will not be committed.